content/tutorials/manage-the-rippled-server/configuration/run-rippled-as-a-validator.md

```diff
@@ -1,74 +1,191 @@
+
+3. [Request domain verification](#request-domain-verification).
+
+
+
+## Enable Validation on Your `rippled` Server
+
+Enabling validation on your `rippled` server means providing a validator token in your server's `rippled.cfg` file. Ripple recommends using the `validator-keys` tool (included in `rippled` yum and Alien RPMs) to securely generate and manage your validator keys and tokens. ***TODO: correct reference to yum and Alien RPMs? These are the only RPMs, correct? Should we also point to https://github.com/ripple/validator-keys-tool for folks who want to manually compile and use it for a validator running on Windows?***
+
+For more information about using the `validator-keys` tool, see the [Validator Keys Tool Guide](https://github.com/ripple/validator-keys-tool/blob/master/doc/validator-keys-tool-guide.md).
+
+1. Generate a validator key pair.
 
         $ /opt/ripple/bin/validator-keys create_keys
 
-    **Warning:** Store the generated `validator-keys.json` key file in a secure but recoverable location, such as an encrypted USB flash drive. Do not modify its contents.
+    **Warning:** Store the generated `validator-keys.json` key file in a secure but recoverable location, such as an encrypted USB flash drive. Do not modify its contents. ***TODO: Is the main point here that this is an offline location? Specifically, is it not secure enough to keep it in the default location it is generated in? Like if I'm running my server in AWS EC2 - should I move the json file to my flash drive and delete it from EC2?***
 
-3. Generate a validator token and edit your `rippled.cfg` file to add the `[validator_token]` value.
+2. Generate a validator token.
 
         $ /opt/ripple/bin/validator-keys create_token --keyfile /path/to/your/validator-keys.json
 
-    If you had previously configured your validator without using the `validator-keys` tool, you must also delete the `[validation_seed]` from your `rippled.cfg` file. This changes your validator public key.
+      Sample output:
+
+        Update rippled.cfg file with these values:
+
+        # validator public key: nHUtNnLVx7odrz5dnfb2xpIgbEeJPbzJWfdicSkGyVw1eE5GpjQr
+
+        [validator_token]
+        eyJ2YWxpZGF0aW9uX3NlY3J|dF9rZXkiOiI5ZWQ0NWY4NjYyNDFjYzE4YTI3NDdiNT
+        QzODdjMDYyNTkwNzk3MmY0ZTcxOTAyMzFmYWE5Mzc0NTdmYT|kYWY2IiwibWFuaWZl
+        c3QiOiJKQUFBQUFGeEllMUZ0d21pbXZHdEgyaUNjTUpxQzlnVkZLaWxHZncxL3ZDeE
+        hYWExwbGMyR25NaEFrRTFhZ3FYeEJ3RHdEYklENk9NU1l1TTBGREFscEFnTms4U0tG
+        bjdNTzJmZGtjd1JRSWhBT25ndTlzQUtxWFlvdUorbDJWMFcrc0FPa1ZCK1pSUzZQU2
+        hsSkFmVXNYZkFpQnNWSkdlc2FhZE9KYy9hQVpva1MxdnltR21WcmxIUEtXWDNZeXd1
+        NmluOEhBU1FLUHVnQkQ2N2tNYVJGR3ZtcEFUSGxHS0pkdkRGbFdQWXk1QXFEZWRGdj
+        VUSmEydzBpMjFlcTNNWXl3TFZKWm5GT3I3QzBrdzJBaVR6U0NqSXpkaXRROD0ifQ==
+
+3. Add `[validator_token]` and its value to your `rippled.cfg` file. ***TODO: Is it okay to also put the "# validator public key: XXXXX" text in the cfg file?***
```

Comment:
Thank you, Rome! I ask b/c the response from the `./validator-keys create_token` command instructs the user to update `rippled.cfg` with the validator public key and the validator_token. See the code block above. Just wanted to be sure that is okay. Absolutely - I know it is a public key -- but I just get a little hypervigilant whenever keys are involved! I need to stop being _unnecessarily_ hypervigilant when it is clearly a public key. 😹 Thank you for humoring me.

I believe this is the master pubkey because it is the `public_key` in the `validator-keys.json` file. Just to be sure I'm keeping this straight - the sub-pubkey would be the `validation_public_key` from `./rippled validation_create`, for example, correct? 🤔

Reply:
1. after looking at the validator-keys-tool docs again, it's definitely the master pubkey
2. The sub-pubkey is encoded into the token. Well, actually, I think it's derived from the sub-private key that's encoded into the token. But the `validation_create` output is just a key pair, not intrinsically "master" or "sub". It depends on how you use it, which if you're using the validator keys tool, you won't use `validation_create` for either one since it kind of does the same thing. (Only the keys tool knows how to relate to keypairs to make one authorize the other.) You might use `validation_create` to make a node keypair since that's in the same format but doesn't support the master key/subkey thing.